raven/processes/regionalization_methods.py

```diff
@@ -0,0 +1,260 @@
+        inputs_file -- The ungauged catchment inputs file to work with
+        number_donors -- The number of gauged catchments to use
+        model_parameters -- Selected hydrological model calibrated parameters
+
+    Function to compute distance only regionalization method.
+    """
+    # Preallocate empty array with final dimensions
+    nc_file = Dataset(inputs_file[0], 'r')
+    nDays=nc_file.variables["rain"][:].shape[0]
+    Qsim = np.zeros(shape=(nDays,number_donors))    
+   
+    
+    for j in range(0, number_donors):
+        
+        params=model_parameters.values[j,2:]
+        
+        if hydrological_model == "GR4JCN":
+            
+            Qs=runLocalGR4JCN(params,ncfilename,start_date, end_date, latitude, area )
+            Qsim[0:Qs.shape[0],j]=Qs
+            
+        elif hydrological_model == "HMETS":
+            
+            Qs=runLocalHMETS(params,ncfilename,start_date, end_date, latitude, area )
+            Qsim[0:Qs.shape[0],j]=Qs    
+            
+        elif hydrological_model == "MOHYSE":
+
+            Qs=runLocalMOHYSE(params,ncfilename,start_date, end_date, latitude, area )
+            Qsim[0:Qs.shape[0],j]=Qs
+            
+        else:
+            print("The hydrological model selected is not supported for regionalization")
+ 
+    Qsim=np.delete(Qsim,list(range(Qs.shape[0],Qsim.shape[0])),axis=0);
+
+    
+    return Qsim
+
+
+def distance_MLR_regionalization(ncfilename,start_date, end_date, latitude, longitude, elevation, area,
+                                 hydrological_model, inputs_file,
+                                 number_donors,
+                                 ungauged_properties,
+                                 model_parameters,
+                                 gauged_properties,
+                                 properties_to_use):
+    """
+    distance_MLR_regionalization
+
+    INPUTS:
+        hydrological_model -- String of the selected hydrological model
+        inputs_file -- The ungauged catchment inputs file to work with
+        number_donors -- The number of gauged catchments to use
+        ungauged_properties -- Properties of the ungauged catchment
+        model_parameters -- Selected hydrological model calibrated parameters
+        gauged_properties -- Properties of all gauged catchments available
+        properties_to_use -- The catchments' properties to use
+
+    Function to compute distance MLR regionalization method.
+    """
+
+    MLR_parameters, r_squared = MLR(ungauged_properties,
+                                    model_parameters,
+                                    gauged_properties,
+                                    properties_to_use)
+
+    model_parameters = model_parameters.values[:, 2:]
+
+    for i in range(0, r_squared.shape[0]):
+        # if we have an R2 > 0.5 then we consider this to be a better estimator
+        if r_squared[i] > 0.5:
+            model_parameters[:, i] = MLR_parameters[i]

```

Comment:
I don't understand this. Why would you replace one parameter by the MLR parameter ? If the r2 is > .5 for all parameters, then you're running the model number_donor times with the same parameter set, no ? If it's not, then you have a mix of parameters from the ungauged catchments and the MLR parameters which won't necessarily work well together.

Reply:
Yes, the idea is that when you transfer a parameter set, you kind of implicitly "decide" that the best available info for each parameter comes from the neighboring/most similar donor. However, if you have some parameters/processes that are more easily identified with a MLR, then this info should be included. It is a philosophical point, some people will say that parameter sets should be left completely intact because the parameters are a set that works together, others will say that each one represents a physical process and should be individually determined from observations, etc. This method (regression-augmentation) is kid of the compromise solution. In this paper [http://richardarsenault.ca/Continuous_streamflow_prediction_ungauged_basins_effect_equifinality.pdf ] we showed that it performed better than all the others in many situations. So for example we could change only 3 parameters out of 21 if those are the only ones that have a certain skill with MLR.

Reply:
Interesting. I'm trying to restructure the code and avoid duplication. I should be finished sometimes today.